app/models/competition.rb

```diff
@@ -382,6 +384,25 @@ def advancement_condition_must_be_present_for_all_non_final_rounds
     end
   end
 
+  validate :auto_accept_validations
+  private def auto_accept_validations
+    errors.add(:auto_accept_registrations, I18n.t('competitions.errors.must_use_wca_registration')) if
+      auto_accept_registrations && !use_wca_registration
+
+    errors.add(:auto_accept_registrations, I18n.t('competitions.errors.auto_accept_limit')) if
+      auto_accept_disable_threshold > 0 &&
+      competitor_limit.present? &&
+      auto_accept_disable_threshold >= competitor_limit
+
+    errors.add(:auto_accept_registrations, I18n.t('competitions.errors.auto_accept_not_negative')) if auto_accept_disable_threshold < 0
+
+    if auto_accept_registrations_changed?
```

Comment:
Why do you need to check this additionally?

Reply:
(i.e. can't you just run the validations inside this `if` globally?)

Reply:
The intention here is to ensure that auto_accept can only be enabled when there are no paid-pending/waitlisted registrations while registration is not full. 

I don't think we want that to run all the time - _after_ auto_accept is enabled there may well be legitimate reasons for an organizer to want to move a paid registration from accepted to pending, and we don't want to break the competition when that happens.

Reply:
- [ ] only enforce this if it changed to `true`